### PR TITLE
Add example notebook for PDP analysis with SageMaker Clarify

### PR DESCRIPTION
**Pull request by oyangz**
*Issue #, if available:* https://issues.amazon.com/issues/P53689304

*Description of changes:*
New Jupyter notebook that shows how to create Partial Dependency Plots (PDP) with Sagemaker Clarify with and without SHAP. 

*Testing done:*
Ran the notebook end to end and confirmed there were no errors.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `black-nb -l 100 {path}/{notebook-name}.ipynb`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by goswamig** [comment]
@larroy

**Comment by goswamig** [comment]
@larroy

**Comment by larroy** [comment]
<p>Minor comments but LGTM</p>

**Review by larroy — APPROVED** [review]
minor comments, LGTM.

File: sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb [hunk 1]
```diff
@@ -0,0 +1,748 @@
...
+   "execution_count": null,
+   "id": "064119ba-6974-4744-9237-dbd14238de6a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "shap_config = clarify.SHAPConfig(\n",
+    "    baseline=[test_features.iloc[0].values.tolist()],\n",
+    "    num_samples=15,\n",
+    "    agg_method=\"mean_abs\",\n",
+    "    save_local_shap_values=True,\n",
```
Comment (bilalaws) [inline sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb:637]: Is this necessary? The only role of SHAP is to select 5 most important features (globally). So we do not need to save local SHAP vals.

File: sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb [hunk 2]
```diff
@@ -0,0 +1,748 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "064119ba-6974-4744-9237-dbd14238de6a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "shap_config = clarify.SHAPConfig(\n",
+    "    baseline=[test_features.iloc[0].values.tolist()],\n",
+    "    num_samples=15,\n",
```
Comment (bilalaws) [inline sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb:635]: Let us remove `num_samples` from here and let the constructor use the default value. If not, we should explain what this parameter does.

File: sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb [hunk 3]
```diff
@@ -0,0 +1,748 @@
...
+    "    grid_resolution=25,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab4392c2-c760-4f62-b317-7c4541a3b84d",
+   "metadata": {},
+   "source": [
+    "The Kernel SHAP algorithm requires a [baseline for explainability](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-feature-attribute-shap-baselines.html), also known as a background dataset. If a `baseline` is not provided, it will be automatically computed as the cluster centers of the dataset by SageMaker Clarify using K-means or K-prototypes. Baseline dataset type shall be the same as `dataset_type` of `DataConfig`, and baseline samples shall only include features. By definition, `baseline` should either be a S3 URI to the baseline dataset file, or an in-place list of samples. In this case we chose the latter, and put the first sample of the test dataset to the list."
```
Comment (bilalaws) [inline sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb:623]: Let us also add: The parameter `agg_method` specifies how the per-instance feature importance should be aggregated over the dataset to compute the overall (global) feature importance. For more details see [Amazon AI Fairness and Explainability Whitepaper](https://pages.awscloud.com/rs/112-TZM-766/images/Amazon.AI.Fairness.and.Explainability.Whitepaper.pdf).

File: sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb [hunk 4]
```diff
@@ -0,0 +1,748 @@
...
+   "id": "219ad306-f7eb-44f4-83fe-01b584065a80",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "clarify_processor.run_explainability(\n",
+    "    data_config=explainability_data_config,\n",
+    "    model_config=model_config,\n",
+    "    explainability_config=[pdp_config, shap_config],\n",
```
Comment (bilalaws) [inline sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb:680]: `logs=False`

**Review by bilalaws — CHANGES_REQUESTED** [review]
Thanks for the PR. Added some suggestions. All except one are about improving readability. The remaining one is about changing a stale hyperlink.

File: sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb [hunk 5]
```diff
@@ -0,0 +1,748 @@
...
+   "execution_count": null,
+   "id": "f9ba7764-8c1b-4314-bbb8-3d2912430c73",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import IPython\n",
+    "\n",
+    "IPython.display.HTML(filename=\"explainability-report.html\")"
```
[the remark this answers is not in the document]
Comment (oyangz) [reply sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb:586]: We are inferring feature types here. I believe currently line plots are the only option for both categorical and numerical features in the report. I will look into this and track it in a separate SIM, and can work on it in another CR.

**Review by bilalaws — APPROVED** [review]

**Review by bilalaws — DISMISSED** [review]

File: sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb [hunk 6]
```diff
@@ -0,0 +1,14468 @@
+{
```
Comment (rdamazon) [inline sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb:1]: <p>can you add plt.show() here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3752/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (rdamazon) [inline sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb:1]: <p>can you add plt.show() here? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3752/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

**Pull request merged**